lib/importproject.cpp

```diff
@@ -232,111 +232,165 @@ static std::string readUntil(const std::string &command, std::string::size_type
     return ret;
 }
 
+static std::string removeBackslash(std::string str)
+{
+    str.erase(std::remove(str.begin(), str.end(), '\\'), str.end());    
+    return str;
+}
+
+static void skip_whitespaces(const std::string& command, std::string::size_type *pos)
+{
+    while (*pos < command.size() && command[*pos] == ' ')
+        (*pos)++;
+}
+
+
+static std::string parseTillNextCommandOpt(const std::string& command, std::string::size_type *pos)
+{
+    std::string ret{};
+    while (*pos < command.size() && command[*pos] != '/' && command[*pos] != '-' )
+    {
+        (*pos)++;
+    }
+    (*pos)++;
+    if (*pos >= command.size())
+    {
+        return ret;
+    }
+    const char F = command[*pos];
+
+    if(std::strchr("DUI", F))
+    {
+      (*pos)++;
+      return std::string{F};
+    }
+
+    ret = readUntil(command, pos, " =");
+    return ret;
+}
+
+void ImportProject::FileSettings::parseCommandStd(const std::string& command, std::string::size_type *pos, std::string& defs)
+{
+    std::string def{};
+
+    const std::string stdval = readUntil(command, pos, " ");
+    standard = stdval;
+    if (standard.compare(0, 3, "c++") || standard.compare(0, 5, "gnu++")) {
+        std::string stddef;
+        if (standard == "c++98" || standard == "gnu++98" || standard == "c++03" || standard == "gnu++03") {
+            stddef = "199711L";
+        } else if (standard == "c++11" || standard == "gnu++11"|| standard == "c++0x" || standard == "gnu++0x") {
+            stddef = "201103L";
+        } else if (standard == "c++14" || standard == "gnu++14" || standard == "c++1y" || standard == "gnu++1y") {
+            stddef = "201402L";
+        } else if (standard == "c++17" || standard == "gnu++17" || standard == "c++1z" || standard == "gnu++1z") {
+            stddef = "201703L";
+        }
+
+        if (stddef.empty()) {
+            // TODO: log error
+        }
+        else
+        {
+          def += "__cplusplus=";
+          def += stddef;
+          def += ";";
+        }
+    }
+
+    defs += def;
+}
+
+void ImportProject::FileSettings::parseCommandDefine(const std::string& command, std::string::size_type *pos, std::string& defs)
+{
+    const std::string fval = readUntil(command, pos, " =");
```

Comment:
isn't it enough to write:

    defs += readUntil(command, pos, " ") + ";";

Reply:
I'll clean it up